From 94bdd8e1c74be624c2f84211b96839dd27275270 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 5 Sep 2017 17:33:50 +0800
Subject: [PATCH 2510/5242] MLK-16372-3 arm64: dts: imx8qxp: add cpu-freq
 cooling map

commit  c84d1646b96e19b06333491e1d90b4613df3c9ca from
https://source.codeaurora.org/external/imx/linux-imx.git

Add i.MX8QXP cpu-freq cooling support, when temperature
exceeds passive point, cpu-freq will drop to lowest
set-point.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index f6c4f3f..929e91d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -21,6 +21,7 @@
 #include <dt-bindings/clock/imx8qxp-clock.h>
 #include <dt-bindings/pinctrl/pads-imx8qxp.h>
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/thermal/thermal.h>
 
 / {
 	compatible = "fsl,imx8qxp";
@@ -831,6 +832,13 @@
 					type = "critical";
 				};
 			};
+			cooling-maps {
+				map0 {
+					trip = <&cpu_alert0>;
+					cooling-device =
+					<&A35_0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
 		};
 	};
 
@@ -2087,4 +2095,5 @@
 	>;
 	clocks = <&clk IMX8QXP_A35_DIV>;
 	clock-latency = <61036>;
+	#cooling-cells = <2>;
 };
-- 
1.7.9.5

